File: conectopia/app/models/Message.scala
package models

import org.hibernate.annotations.{Tuplizer, Target}

import javax.persistence._

@Entity
@Table(name = "message")
class Message extends PersistentEntity {
  // content
  @Column(name = "content")
  var content: String = _
  // timestamp as sql Timestamp
  @Column(name = "timestamp")
  var timestamp: java.sql.Timestamp = _
  // chat_id many to one
  @ManyToOne
  @JoinColumn(name = "chat_id")
  var chat: Chat = _
  // user_id many to one
  @ManyToOne
  @JoinColumn(name = "user_id")
  var user: User = _
  // constructor
  def this(content: String, timestamp: java.sql.Timestamp, chat: Chat, user: User) {
    this()
    this.content = content
    this.timestamp = timestamp
    this.chat = chat
    this.user = user
  }
}

File: play-1.5.3/Conectopia/app/models/Message.java
package models;

import org.hibernate.annotations.Entity;
import play.db.jpa.Model;

import javax.persistence.ManyToOne;
import java.util.Date;

@Entity
public class Message extends Model {
    public String content;
    public Date created;
    @ManyToOne
    public User user;
    @ManyToOne
    public Chat chat;

    public Message(String content, User user, Chat chat) {
        this.content = content;
        this.user = user;
        this.chat = chat;
        this.created = new Date();
    }
public Message() {
        this.content = "";
        this.user = null;
        this.chat = null;
        this.created = new Date();
    }
}
